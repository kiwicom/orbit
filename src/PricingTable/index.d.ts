// @flow
// Type definitions for @kiwicom/orbit-components
// Project: http://github.com/kiwicom/orbit-components

import * as React from "react";

import * as Common from "../common/common";

declare module "@kiwicom/orbit-components/lib/PricingTable";

export interface Props extends Common.Global {
  readonly children: React.ReactNode;
  readonly activeElement?: number | undefined | null;
  readonly hasError?: boolean;
}

export const PricingTable: React.FunctionComponent<Props>;
export { PricingTableItem } from "./PricingTableItem/index";
