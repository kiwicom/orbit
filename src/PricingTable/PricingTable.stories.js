// @flow
import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import Badge from "../Badge";
import Button from "../Button";
import List, { ListItem } from "../List";
import Tooltip from "../Tooltip";
import Text from "../Text";
import Check from "../icons/Check";

import PricingTable, { PricingTableItem } from "./index";

const icon = (
  <img
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAAABGdBTUEAALGPC/xhBQAAB2FJREFUWAm1mHtsFEUcx+fu9nbv2d4dFGppbYTIw6PER2mlYmwtrYDBRyIQAxpfUWNMfJD+4x/ahH/0H0NM1BhFooKaGq0mtgQrXgNIfRQbJC3y0NgWSqW0d7t3t3d7t7vnfKdO2dbS9HhMcjuzOzO/3+f3m99vZvds5ApKy+HD7pAjWO/2eF6VRHFlKqW+kcykP1lbFe69XLG2y524/5fjNV6f/y0KchtkpNNp4vF4SDaTUZNJ9X1lKPbKhg2Var7y7flOwPh9Xb3VgUDgW8BomkaOHj1K2traSH9/PzFM01MYKHxBDEnvtbS0OPKVn7eH2tvbpeD1yw+4XK6qTCZD9u3bR2RZJmjncjmyYsUKsmjRIsaR1Y0ta2vCn+YDlbeHxNDCGqfTWWWaJgM5e/Ysqa6uJqWlpWTjxo1k7ty5RNd1BpfLGU83NzfnpSOvwbDUKTqqKBAxDIP4/X5SUlJCIpEIGRoaIqdPnyY+n48AFv12Ylt2e+PGwDX1kM1u92Bp8BNFkaxbt45UVlaS8+fPszjq6uoiDoeD2Gw2opum09CIeE2BMpnsKSiA0lgsRgYGBkgoFCJlZWWkpqaGnDhxgnkIYwSH4x9bqj+K9myLMNuBfJyZUX/QNN8YXbYQUr21tZXY7XYWN2fOnCGLFy9m3gFwJqN/vX79eo3PnU2ddww1rr51SE0lX9V1Q0csbdq0iYTDYaartraWBThbLt2IC4KzbTYQ1jF5p/13h/4o8Qc8mw3D3J5MJrwQhiBGDCHAEVsAKioqMgVBULKZbA814OOTo39//nhdXdqqfLr2BFBHR3ehEPRX2IjDaeb0oQOJc38219XpfBKOiTJ/6csFhYXPS5JYDMWJRIKMjIwQVR3fkJHubrebzJ8/n3i9XhZnFIrt4vF4/GdZjm1rXLX8Ry5zunoCKPLbqYdKSxZ8AWszmpamedQrx5Vd5xLDH7o10XtdyYIWGrx12AAvXLjAUhw78+joKAMaXyadAECSJEJ3clJeXs7giouLWUYaup6UFeW5u1cu+Xg6GDy7CNR9YnVJyYKD2WyWWYZARWDGFeVnehxkgsHgnQjizs5O0t3dzYIY2TVv3jwyZ84cphDj4S1kH6AHBwdJNBplS7l161a2adKlzsoxeXN99ZLW6aAmsiyZ0nrpaT3sdnuKAcU3N3qSV3NFCGIaG6SiooIsXbqUwHIsDZYPBqDmRVEUtlkeO3aMLRm8hiWlHnT6/L53Og71HmlYHR7g43k94SE8+PX40EeBYOBRHJhQgN0WMLwGJO7RhzYKlOAeBUDohyd5P+/DOCwrZCHO6Pm3886bFz7FJlou45L+exBLRHfQrEhbISCEHqRMGIZBKT9ILwUDpfgBCmPgccjhBmG+S3Jt/v7g7wstLKzJgFoivb72A78V2WT7gJJQdsIqBCdKX18f2b17N3M/P8OswBhj9QwHAQDv4x7lNcZLLskneN0PsEGWi9DZc7IhWBh6mwIEaMBRKwwbtwxnVU9PD9mzZw8L0KamJgbKPQPB+cIAFD+AS6K0hrK8CZ7IkVOviaKzQggGQt9Q97khGAPtdidzMbyAZ5iIN0G8XuAZ4ov3zQYG8zEeRnIYtOEtmgLh9vafClxFgdoCn+8VSjBq6xuUWWrwibzmAhCgw8PDbE+BJRCKcqUwkE+FGXRv+sshCAtdkuQYi8Zes/UOxHLohBVWGChlVlAvIZ54YF4tGC4bNQoMHYuNNtoBcikYAGIgsoJPvBqegSzmIQoCedCR1fW/czH1F3tKTXUieLE38OXABO4tXmPi1YSBLuhB5tIgSKoJtamhoVJ2rN98/1c2Q9Do28R5qnGuy+328eMDMNfCMxwmq2fTmqq9rsSj2+qrl0XY0uHCS9uPfeXXFYU+8BcUrEEwXysYDkRDoev2cFkN14960k597x039Y+cG3kilUqNwpUoV7pM3Ms8awGDNhIlrWmfMSWWyyQgPL/nropB+uW5E0BXCsOzF3IAwmEQs8lk8mRUjn9iYWHN/wHhKX1neUNR4sdxhmHpYCW3EP1QYA183scV8j7U6EPhfeNbiK6rqdSLD9bdEmOdlsu0QPRrcyw6OvYwhRrkQiEQZTYw3DNTQQFDn+Xiifi2+pVL9lo4JprjWiZuJzc6Dh+v8Bd6dzmd4m0QPlsYqxFWz5iGmaSvvU21lTe+O1nTxbtpPcS7G2qWHUultR3YOK0w/1k6KS4wBtCAwViAcBj00YyKRmNja2eCgd6JN0YOMaW2iYKw1QrDX0EAxZXjGWA4OJeB/Yx7SxAcHuIgSd53qXpGoI7DvTeLklgLZQhsZEcmo52TE8ojQs5OiYhvqmCDPrCbdpN+mdxXUFjwGN/PKLTkcrqeod3PTp1jvZ8RSJCcVfQdW8KLOzxCj2YtIStPNq4K77cKma69KxLZu9hxQ7nP62NfKjBKdIqrmyMRwfp5NXXujDGUUJQvZUWO0D8YDDNnphPx5Ev1q8LTZsdUwfgojEfVLal0+ohJ/5dBcqlJdftMMJDxL/yVNn1wgwwEAAAAAElFTkSuQmCC"
    alt=""
  />
);

const content = (
  <List type="separated">
    <ListItem label="Kiwi.com services" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>Extended.</Text>
      </Tooltip>
    </ListItem>
    <ListItem label="Calleer priority" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>Medium</Text>
      </Tooltip>
    </ListItem>
    <ListItem label="Email support" icon={<Check size="small" color="success" />}>
      <Text>Yes</Text>
    </ListItem>
    <ListItem label="Additional Services" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>$10 processing fee</Text>
      </Tooltip>
    </ListItem>
  </List>
);

const longerContent = (
  <List type="separated">
    <ListItem label="Kiwi.com services" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>Extended.</Text>
      </Tooltip>
    </ListItem>
    <ListItem label="Calleer priority" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>Medium</Text>
      </Tooltip>
    </ListItem>
    <ListItem label="Email support" icon={<Check size="small" color="success" />}>
      <Text>Yes</Text>
    </ListItem>
    <ListItem label="Additional Services" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>$10 processing fee</Text>
      </Tooltip>
    </ListItem>
    <ListItem label="Additional Services" icon={<Check size="small" color="success" />}>
      <Tooltip content="Extra info about provided service">
        <Text>$10 processing fee</Text>
      </Tooltip>
    </ListItem>
  </List>
);

storiesOf("PricingTable", module)
  .addDecorator(withKnobs)
  .add(
    "Compact",
    () => {
      return (
        <PricingTable defaultActiveElement={1}>
          <PricingTableItem
            name="Limited Services"
            priceBadge={<Badge type="info">Included</Badge>}
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                type="secondary"
                block
              >
                Don&#39;t upgrade
              </Button>
            }
            mobileDescription="Basic ticket fare includes:"
            onClick={action("onClick")}
          >
            {content}
          </PricingTableItem>
          <PricingTableItem
            name="Plus Services"
            priceBadge={<Badge type="info">+ 10</Badge>}
            badge="Popular"
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                block
              >
                Upgrade and continue
              </Button>
            }
            mobileDescription="Flexi ticket fare includes:"
            onClick={action("onClick")}
          >
            {longerContent}
          </PricingTableItem>
          <PricingTableItem
            name="Premium Services"
            priceBadge={<Badge type="info">+ 20</Badge>}
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                type="secondary"
                block
              >
                Upgrade and continue
              </Button>
            }
            mobileDescription="Premium ticket fare includes:"
            onClick={action("onClick")}
          >
            {content}
          </PricingTableItem>
        </PricingTable>
      );
    },
    {
      info:
        "You can try all possible configurations of this component. However, check Orbit.Kiwi for more detailed design guidelines.",
    },
  )
  .add(
    "With FeatureIcon",
    () => {
      return (
        <PricingTable>
          <PricingTableItem
            name="Basic"
            price="$749"
            featureIcon={icon}
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                block
              >
                Continue with Basic
              </Button>
            }
            mobileDescription="Basic ticket fare includes:"
            onClick={action("onClick")}
          >
            {content}
          </PricingTableItem>
          <PricingTableItem
            name="Plus Services"
            price="$749"
            featureIcon={icon}
            badge={<Badge type="warningInverted">Popular</Badge>}
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                block
              >
                Continue with Basic
              </Button>
            }
            mobileDescription="Flexi ticket fare includes:"
            onClick={action("onClick")}
          >
            {longerContent}
          </PricingTableItem>
          <PricingTableItem
            name="Premium"
            price="$1,095"
            featureIcon={icon}
            action={
              <Button
                onClick={ev => {
                  ev.stopPropagation();
                }}
                block
              >
                Continue with Basic
              </Button>
            }
            mobileDescription="Premium ticket fare includes:"
            onClick={action("onClick")}
          >
            {content}
          </PricingTableItem>
        </PricingTable>
      );
    },
    {
      info:
        "You can try all possible configurations of this component. However, check Orbit.Kiwi for more detailed design guidelines.",
    },
  );
