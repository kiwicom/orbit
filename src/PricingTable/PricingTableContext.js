// @flow
import * as React from "react";

import type { Context } from "./PricingTableContext";

const defaultContext: Context = {
  basis: 0,
  hasError: false,
  desktopRadio: false,
  isContent: false,
};

const PricingTableContext: React.Context<Context> = React.createContext(defaultContext);
PricingTableContext.displayName = "PricingTableOrbitContext";

export default PricingTableContext;
