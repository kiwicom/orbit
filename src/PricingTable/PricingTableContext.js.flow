// @flow

export type Context = {|
  +basis: number,
  +hasError: boolean,
  +desktopRadio: boolean,
  +isContent: boolean,
|};

export type PricingTableContextType = React.Context<Context>;

declare export default PricingTableContextType;
