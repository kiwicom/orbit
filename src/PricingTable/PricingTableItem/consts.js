// @flow
const STATES = {
  HOVER: "hover",
};

export default STATES;
