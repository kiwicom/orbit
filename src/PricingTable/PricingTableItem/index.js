// @flow
import React from "react";
import styled, { css } from "styled-components";

import defaultTheme from "../../defaultTheme";
import Stack from "../../Stack";
import Text from "../../Text";
import Radio from "../../Radio";
import Badge from "../../Badge";
import media from "../../utils/mediaQuery";
import type { Props } from "./index.js.flow";
import STATES from "./consts";

const getBoxShadow = state => ({ theme, active }) => {
  const getActive = array => {
    const copy = array;
    if (active) copy.push(`inset 0 0 0 2px ${theme.orbit.paletteProductNormal}`);
    return copy.join(", ");
  };
  if (state === STATES.HOVER) {
    /* TODO: Add elevation token */
    return getActive(["0 1px 4px 0 rgba(37, 42, 49, 0.16)", "0 4px 8px 0 rgba(37, 42, 49, 0.12)"]);
  }
  return getActive(["0 0 2px 0 rgba(37, 42, 49, 0.16)", "0 1px 4px 0 rgba(37, 42, 49, 0.12)"]);
};
const StyledPricingTableItem = styled.div`
  display: flex;
  flex-grow: 1;
  position: relative;
  background: ${({ theme }) => theme.orbit.paletteWhite};
  box-shadow: ${getBoxShadow()};
  transition: ${({ theme }) => theme.orbit.durationNormal};
  border-radius: ${({ theme }) => theme.orbit.borderRadiusNormal}; /* TODO: Add token */
  padding: ${({ theme }) => theme.orbit.spaceMedium}; /* TODO: Add token */
  cursor: pointer;

  &:hover {
    box-shadow: ${getBoxShadow(STATES.HOVER)};
  }

  ${({ basis }) =>
    basis &&
    css`
      flex-basis: ${basis};
    `}

  ${({ featureIcon }) =>
    featureIcon &&
    css`
      ${media.tablet(css`
        padding-top: ${({ theme }) => theme.orbit.spaceLarge}; /* TODO: Add token */
        padding-bottom: ${({ theme }) => theme.orbit.spaceLarge}; /* TODO: Add token */
      `)}
    `}
`;

StyledPricingTableItem.defaultProps = {
  theme: defaultTheme,
};

const StyledBadge = styled.div`
  position: absolute;
  top: -${({ theme }) => theme.orbit.spaceMedium}; /* TODO: Add token */
  left: 50%;
  transform: translate(-50%, 3px);
`;

StyledBadge.defaultProps = {
  theme: defaultTheme,
};

const Item = styled.div``;

const PricingTableItem = ({
  name,
  price,
  priceBadge,
  featureIcon,
  badge,
  action,
  active = false,
  compact = false,
  children,
  onClick,
  basis,
}: Props) => {
  const onClickHandler = () => {
    if (onClick) {
      onClick();
    }
  };

  return (
    <StyledPricingTableItem
      onClick={onClickHandler}
      basis={basis}
      featureIcon={!!featureIcon}
      active={active}
    >
      {badge && (
        <StyledBadge>
          {typeof badge === "string" ? <Badge type="infoInverted">{badge}</Badge> : badge}
        </StyledBadge>
      )}
      <Stack flex direction="column" spacing="condensed" tablet={{ spacing: "natural" }}>
        {featureIcon && (
          <Stack justify="center" grow={false}>
            {featureIcon}
          </Stack>
        )}
        <Stack justify="between" direction="column">
          <Stack spacing="tight" direction="column" flex align="stretch" tablet={{ grow: false }}>
            {name && (
              <Text type="primary" align="center" weight={featureIcon ? "normal" : "bold"}>
                {name}
              </Text>
            )}
            {price && (
              <Text size="large" weight="bold" type="primary" align="center">
                {price}
              </Text>
            )}
            {priceBadge && (
              <Stack justify="center" align="end" tablet={{ grow: false }}>
                {priceBadge}
              </Stack>
            )}
          </Stack>
          {!compact ? (
            <Stack justify="between" direction="column">
              {children && children}
              {action && (
                <Stack justify="center" grow={false}>
                  {action}
                </Stack>
              )}
            </Stack>
          ) : (
            <Stack justify="center" align="center" grow={false}>
              <Item>
                <Radio checked={active} onChange={onClickHandler} />
              </Item>
            </Stack>
          )}
        </Stack>
      </Stack>
    </StyledPricingTableItem>
  );
};

export default PricingTableItem;
