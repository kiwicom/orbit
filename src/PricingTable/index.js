// @flow
import React, { useState } from "react";
import styled from "styled-components";

import Stack from "../Stack";
import Text from "../Text";
import useMediaQuery from "../hooks/useMediaQuery";
import type { Props } from "./index.js.flow";

const StyledPricingTable = styled.div``;

const PricingTable = ({ children, defaultActiveElement = 0, dataTest }: Props) => {
  const { isDesktop } = useMediaQuery();
  const [activeElement, setActiveElement] = useState(defaultActiveElement);
  const handleOnClick = i => {
    return () => {
      if (!isDesktop) {
        setActiveElement(i);
      }
    };
  };
  const resolveBasis = item => {
    if (item.length) {
      return `${Math.floor(100 / item.length)}%`;
    }
    return `100%`;
  };
  return (
    <>
      {isDesktop !== null && (
        <StyledPricingTable data-test={dataTest}>
          <Stack
            flex
            grow
            spaceAfter="medium"
            spacing="condensed"
            align="stretch"
            desktop={{ spacing: "natural", spaceAfter: "none" }}
            justify="center"
          >
            {isDesktop
              ? children
              : React.Children.map(children, (child, i) =>
                  React.cloneElement(child, {
                    active: activeElement === i,
                    compact: true,
                    basis: resolveBasis(child),
                    onClick: handleOnClick(i),
                  }),
                )}
          </Stack>
          {!isDesktop && children && (
            <Stack spacing="condensed">
              {React.Children.map(children, (child, i) => {
                if (i === activeElement) {
                  return (
                    <>
                      {child.props.mobileDescription && (
                        <Text weight="bold" size="normal">
                          {child.props.mobileDescription}
                        </Text>
                      )}
                      {child.props.children && React.cloneElement(child.props.children)}
                      {child.props.action && React.cloneElement(child.props.action)}
                    </>
                  );
                }
                return null;
              })}
            </Stack>
          )}
        </StyledPricingTable>
      )}
    </>
  );
};

export default PricingTable;
export { default as PricingTableItem } from "./PricingTableItem";
