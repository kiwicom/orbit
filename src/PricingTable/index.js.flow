// @flow
/*
  DOCUMENTATION: https://orbit.kiwi/components/PricingTable/
*/

import typeof PricingTableItemType from "./PricingTableItem";
import type { Globals } from "../common/common.js.flow";

export type Props = {|
  ...Globals,
  +children: React$Node, // TODO: Type this propperly
  +defaultActiveElement?: number,
|};

declare export default React$ComponentType<Props>;

declare export var PricingTableItem: PricingTableItemType;
