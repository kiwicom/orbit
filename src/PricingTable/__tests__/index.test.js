// @flow

import * as React from "react";
import { shallow } from "enzyme";

import Button from "../../Button";
import Badge from "../../Badge";
import PricingTableItem from "../PricingTableItem";
import PricingTable from "../index";

const onClick = jest.fn();
const onClick2 = jest.fn();
const dataTest = "test";
const name = "name";
const price = "$749";
const actions = <Button>Buy</Button>;
const content = <ul>List</ul>;
const image = <img alt="" />;
const badge = <Badge>Badge</Badge>;

jest.mock("../../hooks/useMediaQuery", () => {
  return () => {
    return {
      isTablet: false,
    };
  };
});

describe("PricingTableItem mobile", () => {
  const component = shallow(
    <PricingTableItem
      dataTest={dataTest}
      name={name}
      price={price}
      featureIcon={image}
      badge={badge}
      action={actions}
      active
      compact
      onClick={onClick}
    >
      {content}
    </PricingTableItem>,
  );
  const pricingTableItem = component.find("PricingTableItem__StyledPricingTableItem");
  it("shouldn't render children", () => {
    expect(component.find("ul").exists()).toBe(false);
  });
  it("should have name", () => {
    expect(
      pricingTableItem
        .find("Text")
        .at(0)
        .children()
        .text(),
    ).toBe(name);
  });
  it("should render data-test", () => {
    expect(pricingTableItem.render().prop("data-test")).toBe(dataTest);
  });
  it("should have price", () => {
    expect(
      pricingTableItem
        .find("Text")
        .at(1)
        .children()
        .text(),
    ).toBe(price);
  });
  it("should have featureIcon", () => {
    expect(pricingTableItem.find("img").exists()).toBe(true);
  });
  it("should have badge", () => {
    expect(pricingTableItem.find("PricingTableItem__StyledBadgeWrapperContent").exists()).toBe(
      true,
    );
  });
  it("shouldn't have actions", () => {
    expect(pricingTableItem.find("Button").exists()).toBe(false);
  });
  it("should have Radio checked radio button", () => {
    expect(
      pricingTableItem
        .find("Radio")
        .render()
        .prop("checked"),
    ).toBe(true);
  });
  it("should be actionable", () => {
    pricingTableItem.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});

describe("PricingTableItem desktrop", () => {
  const component = shallow(
    <PricingTableItem
      dataTest={dataTest}
      name={name}
      price={price}
      featureIcon={image}
      badge={badge}
      action={actions}
      active
      onClick={onClick2}
    >
      {content}
    </PricingTableItem>,
  );
  const pricingTableItem = component.find("PricingTableItem__StyledPricingTableItem");
  it("should render children", () => {
    expect(component.find("ul").exists()).toBe(true);
  });
  it("should have name", () => {
    expect(
      pricingTableItem
        .find("Text")
        .at(0)
        .children()
        .text(),
    ).toBe(name);
  });
  it("should have price", () => {
    expect(
      pricingTableItem
        .find("Text")
        .at(1)
        .children()
        .text(),
    ).toBe(price);
  });
  it("should have featureIcon", () => {
    expect(pricingTableItem.find("img").exists()).toBe(true);
  });
  it("should have badge", () => {
    expect(pricingTableItem.find("PricingTableItem__StyledBadgeWrapperContent").exists()).toBe(
      true,
    );
  });
  it("should have actions", () => {
    expect(pricingTableItem.find("Button").exists()).toBe(true);
  });
  it("shouldn't have Radio button", () => {
    expect(pricingTableItem.find("Radio").exists()).toBe(false);
  });
  it("should be actionable", () => {
    pricingTableItem.simulate("click");
    expect(onClick2).toHaveBeenCalled();
  });
});

describe("PricingTable", () => {
  const mobileDescription = "Basic ticket fare includes:";
  const component = shallow(
    <PricingTable dataTest={dataTest}>
      <PricingTableItem
        dataTest={dataTest}
        name={name}
        price={price}
        featureIcon={image}
        badge={badge}
        action={actions}
        active
        onClick={onClick2}
        mobileDescription={mobileDescription}
      >
        {content}
      </PricingTableItem>
      <PricingTableItem
        dataTest={dataTest}
        name={name}
        price={price}
        featureIcon={image}
        badge={badge}
        action={actions}
        active
        onClick={onClick2}
      >
        {content}
      </PricingTableItem>
      <PricingTableItem
        dataTest={dataTest}
        name={name}
        price={price}
        featureIcon={image}
        badge={badge}
        action={actions}
        active
        onClick={onClick2}
      >
        {content}
      </PricingTableItem>
    </PricingTable>,
  );
  it("should render data-test", () => {
    expect(component.render().prop("data-test")).toBe(dataTest);
  });
  it("should render children", () => {
    expect(component.find("PricingTableItem")).toHaveLength(3);
  });
  it("should render childrens actions", () => {
    expect(component.find("Button").exists()).toBe(true);
  });
  it("should render childrens children", () => {
    expect(component.find("ul").exists()).toBe(true);
  });
  it("should render childrens mobileDescription", () => {
    expect(
      component
        .find("Text")
        .at(0)
        .children()
        .text(),
    ).toBe(mobileDescription);
  });
});
