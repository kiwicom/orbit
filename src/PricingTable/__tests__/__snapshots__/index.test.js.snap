// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PricingTable should match snapshot 1`] = `
<PricingTable__StyledPricingTable>
  <Stack
    align="stretch"
    desktop={
      Object {
        "spaceAfter": "none",
        "spacing": "natural",
      }
    }
    flex={true}
    grow={true}
    spaceAfter="medium"
    spacing="condensed"
  >
    <PricingTableItem
      action={
        <ForwardRef>
          Buy
        </ForwardRef>
      }
      active={true}
      badge={
        <Badge>
          Badge
        </Badge>
      }
      basis="100%"
      compact={true}
      dataTest="test"
      featureIcon={
        <img
          alt=""
        />
      }
      key=".0"
      mobileDescription="Basic ticket fare includes:"
      name="name"
      onClick={[Function]}
      price="$749"
    >
      <ul>
        List
      </ul>
    </PricingTableItem>
    <PricingTableItem
      action={
        <ForwardRef>
          Buy
        </ForwardRef>
      }
      active={false}
      badge={
        <Badge>
          Badge
        </Badge>
      }
      basis="100%"
      compact={true}
      dataTest="test"
      featureIcon={
        <img
          alt=""
        />
      }
      key=".1"
      name="name"
      onClick={[Function]}
      price="$749"
    >
      <ul>
        List
      </ul>
    </PricingTableItem>
    <PricingTableItem
      action={
        <ForwardRef>
          Buy
        </ForwardRef>
      }
      active={false}
      badge={
        <Badge>
          Badge
        </Badge>
      }
      basis="100%"
      compact={true}
      dataTest="test"
      featureIcon={
        <img
          alt=""
        />
      }
      key=".2"
      name="name"
      onClick={[Function]}
      price="$749"
    >
      <ul>
        List
      </ul>
    </PricingTableItem>
  </Stack>
  <Stack
    spacing="condensed"
  >
    <Text
      size="normal"
      weight="bold"
    >
      Basic ticket fare includes:
    </Text>
    <ul>
      List
    </ul>
    <Button>
      Buy
    </Button>
  </Stack>
</PricingTable__StyledPricingTable>
`;
